core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -522,10 +522,14 @@ private[spark] class SparkSubmit extends Logging {
       OptionAssigner(args.queue, YARN, ALL_DEPLOY_MODES, confKey = "spark.yarn.queue"),
       OptionAssigner(args.numExecutors, YARN, ALL_DEPLOY_MODES,
         confKey = "spark.executor.instances"),
-      OptionAssigner(args.pyFiles, YARN, ALL_DEPLOY_MODES, confKey = "spark.yarn.dist.pyFiles"),
-      OptionAssigner(args.jars, YARN, ALL_DEPLOY_MODES, confKey = "spark.yarn.dist.jars"),
-      OptionAssigner(args.files, YARN, ALL_DEPLOY_MODES, confKey = "spark.yarn.dist.files"),
-      OptionAssigner(args.archives, YARN, ALL_DEPLOY_MODES, confKey = "spark.yarn.dist.archives"),
+      OptionAssigner(args.jars, YARN, ALL_DEPLOY_MODES, confKey = "spark.yarn.dist.jars",
+        mergeFn = Some(mergeFileLists(_, _))),
```

Comment:
nit. Just a question. Why this is different from the original patch?

The original patch didn't change the order of lines here. So, `pyFiles` should be at the first instead of `jars`.

Reply:
Oops, this got it while resolving a conflict with cherry-pick (`spark.executor.instances` was removed in branch-3). I've put it back in order (although I believe it should not affect anything).

Reply:
Thanks!